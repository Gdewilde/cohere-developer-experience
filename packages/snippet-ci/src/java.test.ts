import { describe, expect, it } from '@jest/globals';
import { exec } from 'child_process';
import { globSync } from 'glob';
import * as path from 'path';
import * as util from "util";

const execAsync = util.promisify(exec);


describe("java snippets", () => {
    const javaFiles = globSync(path.resolve('../../snippets/java/app/src/**/*.java'), { absolute: false })

    it("file snapshot", async () => {
        expect(javaFiles).toMatchSnapshot();
    });


    it.concurrent.each(javaFiles.slice(0, 2))("java file %s", async (javaFile) => {
        const [javaProjectRoot, javaPath] = javaFile.split("app/src/main/java/")
        const className = javaPath?.split("/").join(".").replace('.java', '');

        const { stdout, stderr } = await execAsync(
            `./gradlew -q -PmainClass=${className} run`,
            {
                env: {
                    "CO_API_KEY": "xx"
                },
                cwd: javaProjectRoot
            });

        console.log('stdout:', JSON.parse(stdout));
        console.log('stderr:', stderr);
    })
});
