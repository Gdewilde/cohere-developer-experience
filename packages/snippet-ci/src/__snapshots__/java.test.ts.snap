// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`java snippets file snapshot 1`] = `
[
  "../../snippets/java/app/src/test/java/AppTest.java",
  "../../snippets/java/app/src/main/java/TokenizePost.java",
  "../../snippets/java/app/src/main/java/SummarizePost.java",
  "../../snippets/java/app/src/main/java/RerankPost.java",
  "../../snippets/java/app/src/main/java/ModelsListGet.java",
  "../../snippets/java/app/src/main/java/GeneratePost.java",
  "../../snippets/java/app/src/main/java/EmebedJobsGet.java",
  "../../snippets/java/app/src/main/java/EmbedPost.java",
  "../../snippets/java/app/src/main/java/EmbedJobsPost.java",
  "../../snippets/java/app/src/main/java/EmbedJobsGet.java",
  "../../snippets/java/app/src/main/java/EmbedJobsCancel.java",
  "../../snippets/java/app/src/main/java/DetokenizePost.java",
  "../../snippets/java/app/src/main/java/DatasetUsageGet.java",
  "../../snippets/java/app/src/main/java/DatasetPost.java",
  "../../snippets/java/app/src/main/java/DatasetGet.java",
  "../../snippets/java/app/src/main/java/DatasetDelete.java",
  "../../snippets/java/app/src/main/java/ConnectorsList.java",
  "../../snippets/java/app/src/main/java/ConnectorsIdOauthAuthorizePost.java",
  "../../snippets/java/app/src/main/java/ConnectorPatch.java",
  "../../snippets/java/app/src/main/java/ConnectorGet.java",
  "../../snippets/java/app/src/main/java/ConnectorDelete.java",
  "../../snippets/java/app/src/main/java/ConnectorCreate.java",
  "../../snippets/java/app/src/main/java/ClassifyPost.java",
  "../../snippets/java/app/src/main/java/finetuning/UpdateFinetunedModel.java",
  "../../snippets/java/app/src/main/java/finetuning/ListTrainingStepMetrics.java",
  "../../snippets/java/app/src/main/java/finetuning/ListFinetunedModels.java",
  "../../snippets/java/app/src/main/java/finetuning/ListEvents.java",
  "../../snippets/java/app/src/main/java/finetuning/GetFinetunedModel.java",
  "../../snippets/java/app/src/main/java/finetuning/DeleteFinetunedModel.java",
  "../../snippets/java/app/src/main/java/finetuning/CreateFinetunedModel.java",
  "../../snippets/java/app/src/main/java/chatpost/Tools.java",
  "../../snippets/java/app/src/main/java/chatpost/Stream.java",
  "../../snippets/java/app/src/main/java/chatpost/Documents.java",
  "../../snippets/java/app/src/main/java/chatpost/Default.java",
]
`;
